### DISCUSSION TITLE
Benchmark AoT OptimizationPreference

### DISCUSSION THREAD

**Discussion by eNeRGy164** [dotnet/BenchmarkDotNet#2496] [Q&A] [answered]
With AoT 8.0 we can define the MSBuild property `OptimizationPreference` as `Speed` or `Size`.
<https://learn.microsoft.com/en-us/dotnet/core/deploying/native-aot/optimizing>

I would like to run benchmarks to see the difference (Clock Wall tests with `StopWatch` shows 8% difference in some of my apps)

How can I create Jobs with different values for this specific setting?

**Comment by timcassell** [comment]
Something like this.

```cs
var config = DefaultConfig.Instance
    .AddJob(Job.Default.WithArguments(new Argument[] { new MsBuildArgument("/p:OptimizationPreference=Speed") }).WithId("Speed"))
    .AddJob(Job.Default.WithArguments(new Argument[] { new MsBuildArgument("/p:OptimizationPreference=Size") }).WithId("Size"));
BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args, config);
```

**Reply by adamsitnik** [reply, marked as answer]
You can also use the `NativeAotToolchainBuilder.IlcOptimizationPreference` and build a dedicated toolchain with different settings

https://github.com/dotnet/BenchmarkDotNet/blob/7dacb16249714932b94c60257951ef9e4371f7d5/src/BenchmarkDotNet/Toolchains/NativeAot/NativeAotToolchainBuilder.cs#L124
